torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -4992,6 +5038,17 @@ def test_post_localSGD_optimizer_parity_with_hierarchical_sgd_grad_is_view(self)
                 grad_is_view=True,
             )
 
+        @skip_if_lt_x_gpu(2)
+        @sandcastle_skip_if(
+            BACKEND not in DistTestCases.backend_feature["ddp"],
+            f"The {BACKEND} backend does not support DistributedDataParallel"
+        )
+        def test_post_localSGD_optimizer_step_reload(self):
```

Comment:
lets also have a unittest for the case we discussed offline - loading a checkpoint that does not have `step`.

For this you can probably take a checkpoint, remove `step` from the state dict, save that, reload, validate etc

Reply:
Thanks for suggestion! Updated unittest accordingly